Adds the possibility to alternate between S0 and S3 or S0 and S4.

jloehel opened the pull request:
Signed-off-by: Jürgen Löhel <jloehel@suse.com>

Commit:
Adds the possibility to alternate between S0 and S3 or S0 and S4.

Signed-off-by: Jürgen Löhel <jloehel@suse.com>

sndirsch commented on stsx:
```diff
@@ -312,14 +316,14 @@ while [ $c -lt $maxnum ] ; do
+    alternate=$method
+    method=$temp
+    resume=20
+    if [ "$method" =~ "^s4k?$" ] ; then
```
I believe this should be 

  if [[ "$method" =~ ^s4 ]]; then

instead. Right?

jloehel replied:
This regex matches only for the strings s4 and s4k. But you are right the script checks the methods before the regex and I have only this set of methods ["s4", "s4k", "s3", "s3k", "s0", "s0k"] for alternate. So the script only needs to check if the string starts with "s4". Thank you for the hint. I will change it.

sndirsch replied:
Sure you've tested this?

#!/bin/bash
method=s4k
if [ "$method" =~ "^s4" ]; then
  echo hello
fi

gives me

./test.sh: line 5: [: =~: binary operator expected

Whereas

#!/bin/bash
method=s4k
if [[ "$method" =~ ^s4 ]]; then
  echo hello
fi

gives me

hello

sndirsch reviewed (commented):
I found one issue ...

Commit:
Makes the regex in the alternate mode a bit simpler.

The regex matched for the strings "s4" and "s4k". But the script checks the methods
before the regex and the regex checks only this set of methods
   ["s4", "s4k", "s3", "s3k", "s0", "s0k"]
for alternate. So the script only needs to check if the string starts with "s4".

Signed-off-by: Jürgen Löhel <jloehel@suse.com>

sndirsch reviewed (commented):
See latest comment ...

Commit:
Replaces regex with posix compatible expression.

Signed-off-by: Jürgen Löhel <jloehel@suse.com>

jloehel commented:
You are right again. I fixed it.
